src/pq_flash_index.cpp

```diff
@@ -1059,6 +1059,21 @@ void PQFlashIndex<T, LabelT>::cached_beam_search(const T *query1, const uint64_t
                                                  const uint32_t io_limit, const bool use_reorder_data,
                                                  QueryStats *stats)
 {
+    IndexSearchContext<LabelT> context(/*time_limit_in_microseconds*/ 0u, io_limit);
+    context.SetLabel(filter_label, use_filter);
+    return cached_beam_search(query1, k_search, l_search, indices, distances, beam_width, context, use_reorder_data,
+                              stats);
+}
+
+template <typename T, typename LabelT>
+void PQFlashIndex<T, LabelT>::cached_beam_search(const T *query1, const uint64_t k_search, const uint64_t l_search,
+                                                 uint64_t *indices, float *distances, const uint64_t beam_width,
+                                                 IndexSearchContext<LabelT> &context, const bool use_reorder_data,
+                                                 QueryStats *stats)
+{
+    auto use_filter = context.UseFilter();
```

Comment:
As much as I hate to pick nits, we do seem to use lower-snake-case for all methods and functions.  We should probably stick with that convention and/or make everything Pascal